provider/datagen/src/bin/datagen.rs

```diff
@@ -112,6 +112,12 @@ fn main() -> eyre::Result<()> {
                 )
                 .default_value("small"),
         )
+        .arg(
+            Arg::with_name("UNIHAN")
+                .long("unihan")
+                .takes_value(false)
+                .help("Use unihan collation data.")
```

Comment:
Discussion: Not sure if I like the boolean. Maybe we change the default in the future, for example.

I think I prefer `--collation-han-mode=unihan` over `--unihan`

Thoughts? @Manishearth @zbraniecki

Reply:
This should use the same default as `SourceData`, so if we change the default we need to change it everywhere. Then on `SourceData` we could either
* add a new method `with_implicithan()` and deprecate `with_unihan()` (because it becomes a no-op)
* or now change `with_unihan()` to `with_unihan(bool)` and change the default bool value
* or have a with_han_collation_mode(HanCollationMode) which would also add an enum to the API

Reply:
No strong opinions. I think modeswitches are overall better though, as opposed to booleans.

Reply:
I prefer adding an enum (which should be non-exhaustive of course).

Let's actually not call it "mode" because that's confusing with the other Han collation runtime flags like "stroke", "pinyin", etc. I like:

- Command line: `--collation-han-database=unihan` and `--collation-han-database=implicithan`
- API: `with_collation_han_database(choice: CollationHanDatabase)`

@markusicu wdyt?

Reply:
`--coll-root=unihan|implicithan`

Reply:
Cool. I'm happy with Markus's proposal for the command-line version, and for the API version, `with_collation_root(root: CollationRoot)` with `enum CollationRoot { UniHan, ImplicitHan }`